hw/drivers/chg_ctrl/da1469x_charger/src/da1469x_charger_shell.c

```diff
@@ -0,0 +1,970 @@
+        i_precharge = (uint32_t)parse_ll_bounds(argv[2], 1, 56, &rc);
+        if (rc) {
+            console_printf("I_PRECHARGE should be in range 1-56\n");
+            return 0;
+        }
+        val &= ~CHARGER_CHARGER_CURRENT_PARAM_REG_I_PRECHARGE_Msk;
+        val |= DA1469X_ENCODE_PRECHG_I(i_precharge);
+    }
+    if (argc > 3) {
+        i_eoc = (uint32_t)parse_ll_bounds(argv[3], 4, 35, &rc);
+        if (rc) {
+            console_printf("I_EOC should be in range 4-35\n");
+            return 0;
+        }
+        val &= ~(CHARGER_CHARGER_CURRENT_PARAM_REG_I_END_OF_CHARGE_Msk |
+                 CHARGER_CHARGER_CURRENT_PARAM_REG_I_EOC_DOUBLE_RANGE_Msk);
+        val |= DA1469X_ENCODE_EOC_I(i_eoc);
+    }
+    CHARGER->CHARGER_CURRENT_PARAM_REG = val;
+
+    return 0;
+}
+
+static int
+da1469x_charger_shell_cmd(int argc, char **argv)
+{
+    if (argc == 1) {
+        return da1469x_charger_help();
+    }
+
+    argv++;
+    argc--;
+    if (strcmp(argv[0], "help") == 0) {
+        return da1469x_charger_help();
+#if MYNEWT_VAL(DA1469X_CHARGER_CLI_DECODE)
+    } else if (strcmp(argv[0], "decode") == 0) {
+        return da1469x_charger_shell_cmd_decode(argc, argv);
+#endif
+    } else if (strcmp(argv[0], "dump") == 0) {
+        return da1469x_charger_shell_cmd_dump(argc, argv);
+    } else if (strcmp(argv[0], "enable") == 0) {
+        return da1469x_charger_shell_enable(argc, argv);
+    } else if (strcmp(argv[0], "disable") == 0) {
+        return da1469x_charger_shell_disable(argc, argv);
+    } else if (strcmp(argv[0], "status") == 0) {
+        return da1469x_charger_shell_charger_status(argc, argv);
+    }
+
+    return da1469x_charger_shell_err_unknown_arg(argv[1]);
+}
+
+static const struct shell_cmd da1469x_charger_shell_cmd_struct = {
+    .sc_cmd = "charger",
+    .sc_cmd_func = da1469x_charger_shell_cmd
+};
+
+
+static int
+da1469x_charger_reg_cmd(int argc, char **argv)
+{
+    const struct reg *reg = NULL;
+    uint32_t val;
+    int status;
+    bool dump = false;
+    bool decode;
+#if MYNEWT_VAL(DA1469X_CHARGER_CLI_DECODE)
+    decode = decode_fields;
+#endif
+
+    for (int i = 0; i < NUM_CHARGER_REGS; ++i) {
+        if (strcasecmp(charger_regs[i].name, argv[0]) == 0) {
+            reg = charger_regs + i;
+        }
+    }
+    assert(reg);
+    if (!reg) {
+        return 0;
+    }
+
+    if (argc == 1) {
+        dump = true;
+#if MYNEWT_VAL(DA1469X_CHARGER_CLI_DECODE)
+    } else if (strcmp(argv[1], "decode") == 0) {
+        dump = true;
+        decode = true;
+#endif
+    } else {
+        val = (uint32_t)parse_ull_bounds(argv[1], 0, 0xFFFFFFFF, &status);
+        if (status) {
+            console_printf("Invalid register value %s\n", argv[1]);
+            return 0;
+        }
+
+    }
+    if (dump) {
+        val = *reg->addr;
+        dump_register(reg, val, decode);
+    } else {
+        *reg->addr = val;
+    }
+
+    return 0;
+}
+
+static const struct shell_cmd da1469x_charger_cmds[] =
+    {
```

Comment:
Coding style nit - I'd move this open brace to same line as da1469x_charger_cmds[].